projects/kit/components/range/range.component.ts

```diff
@@ -167,10 +173,31 @@ export class TuiRange extends TuiControl<[number, number]> implements OnChanges
     }
 
     private updateStart(value: number): void {
-        this.onChange([Math.min(value, this.value()[1]), this.value()[1]]);
+        const newValue = Math.min(value, this.value()[1]);
+        const distance = this.value()[1] - newValue;
+
+        if (this.isDistanceExceeded(distance)) {
+            return;
+        }
+
+        this.onChange([newValue, this.value()[1]]);
     }
 
     private updateEnd(value: number): void {
-        this.onChange([this.value()[0], Math.max(value, this.value()[0])]);
+        const newValue = Math.max(value, this.value()[0]);
+        const distance = newValue - this.value()[0];
+
+        if (this.isDistanceExceeded(distance)) {
+            return;
+        }
+
+        this.onChange([this.value()[0], newValue]);
+    }
+
+    private isDistanceExceeded(distance: number): boolean {
+        return (
```

Comment:
Can we write `tuiClamp(distance, this.limit, this.margin) === distance`? I think it looks easier to read.

Reply:
Oh, sure! It looks better. I didn't know about tuiClamp function. Fixed